### PR TITLE
Add methods to check if values were included in the AIS messages.

### PR DESCRIPTION
**Pull request by jyrkija**
My main issue was that ROT conversion loses null-values. 
According to https://www.navcen.uscg.gov/?pageName=AISMessagesA : "-128 (80 hex) indicates no turn information available (default)."
AISPositionParser uses RateOfTurn.toDegreesPerMinute which converts this to 0, and there is no way to know if the value was missing, or if it was actually 0. 

Other fields also have values for "undefined", but these are not lost since the conversion is a plain multiplication/division. However it would be better to not have to check for magic numbers in the parsed results.

In Java8, you could use Optional<Double> and I guess nullable Doubles would work too, but that would be a destructive change for current users. Maybe "boolean isXAvailable()" would be acceptable? Here's a pull request along those lines, though it's regrettably a bit large.

While I was at it, my compulsive refactoring -disorder struck and I moved the latitude/longitude validity checks from PositionInfo into their respective classes and made them use original int-values rather than converted doubles (http://www.java67.com/2015/09/float-and-double-value-comparison-in-java-use-relational.html).

### CODE REVIEW

File: src/main/java/net/sf/marineapi/ais/message/AISUTCReport.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
  * 
  * @author Lázár József
  */
-public interface AISUTCReport extends AISMessage {
+public interface AISUTCReport extends AISMessage, AISPositionInfo {
```
Comment (ktuukkan) [inline src/main/java/net/sf/marineapi/ais/message/AISUTCReport.java]: ATM, there may not be UTC messages without the position, but I think it would still be safer design to keep the UTC and position interfaces separate, thus allowing to use them independently.

File: src/main/java/net/sf/marineapi/ais/message/AISPositionInfo.java [hunk 2]
```diff
@@ -36,9 +36,21 @@
...
 	/**
 	 * Returns the latitude in degrees.
 	 */
 	double getLatitudeInDegrees();
+
+	/**
+	 * Returns true if longitude is available in the message.
+	 * If false, getLongitudeInDegrees may return an out-of-range value.
+	 */
+	boolean isLongitudeAvailable();
```
Comment (ktuukkan) [inline src/main/java/net/sf/marineapi/ais/message/AISPositionInfo.java]: I'd suggest to use "has" instead of "is" for shorter notation and readability, e.g. in this case `hasLongitude()`.

File: src/main/java/net/sf/marineapi/ais/util/SpeedOverGround.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+package net.sf.marineapi.ais.util;
+
+import java.text.DecimalFormat;
+
+/**
+ * Checks a speed-over-ground value for validity.
+ *
+ * @author Jyrki Jakobsson
+ */
+public class SpeedOverGround {
```
Comment (ktuukkan) [inline src/main/java/net/sf/marineapi/ais/util/SpeedOverGround.java]: Maybe we should add separate tests for utils like this, too. Not a big isssue regarding this PR but currently the utils are only tested through messages, thus leaving some parts of code or edge values uncovered.

File: src/main/java/net/sf/marineapi/ais/message/AISUTCReport.java (hunk 1 above)
Comment (jyrkija) [reply src/main/java/net/sf/marineapi/ais/message/AISUTCReport.java]: Ok, I'll change it back. Did this because the actual fields are in the UTC message implementation.

File: src/main/java/net/sf/marineapi/ais/message/AISPositionInfo.java (hunk 2 above)
Comment (jyrkija) [reply src/main/java/net/sf/marineapi/ais/message/AISPositionInfo.java]: Check.

File: src/main/java/net/sf/marineapi/ais/util/SpeedOverGround.java (hunk 3 above)
Comment (jyrkija) [reply src/main/java/net/sf/marineapi/ais/util/SpeedOverGround.java]: I added a few tests for this and the Latitude & Longitude classes. Pretty trivial, but at least they document the difference between isCorrect and isAvailable.
